src/units.cpp

```diff
@@ -427,7 +427,16 @@ using UnitsMap = std::map<std::string, double>;
 
 bool updateUnitsMap(const UnitsPtr &units, UnitsMap &unitsMap, double exp = 1.0)
 {
-    if (units->isBaseUnit()) {
+    if (isStandardUnitName(units->name())) {
+        auto unitsListIter = standardUnitsList.find(units->name());
+        for (const auto &baseUnitsComponent : unitsListIter->second) {
+            auto unitsMapIter = unitsMap.find(baseUnitsComponent.first);
+            if (unitsMapIter == unitsMap.end()) {
+                unitsMap[baseUnitsComponent.first] = 0.0;
+            }
+            unitsMap[baseUnitsComponent.first] += baseUnitsComponent.second * exp;
+        }
+    } else if (units->isBaseUnit()) {
```

Comment:
I would swap the `isBaseUnit()` and `isStandardUnitName()` tests. It shouldn't make much difference in terms of speed, just that it would be more logical to me. :)

Reply:
Alas, that's what caused the bug in the first place, so the order has to stay :)

Reply:
Hmm... indeed, I can see that now. Bad data structure, but nevermind...

Reply:
It's partly because of the isBaseUnit function I think ... I thought that was simply a test for the SI units, or custom ones, but it's looking at imports instead.  This is where the bug *really* came from - because the `volt` had no children and was not imported, it was wrongly being called a baseUnit.  The effects of it do need to be outside that function though, which is why I've put the base unit update for standard units as the fix, rather than changing that function.